tools/custom-checks/run-attw-on-tgz.ts

```diff
@@ -0,0 +1,44 @@
+import esMain from "es-main";
+import { getAllTgzPath } from "./get-all-tgz-path";
+import { exec } from "child_process";
+import { exit } from "process";
+
+let attwFailedPackages: string[] = [];
+
+export async function runAttwOnTgz(): Promise<void> {
+  const TAG = "[tools/custom-checks/run-attw-on-tgz.ts]";
+  console.log(`${TAG} Fetching .tgz file paths.`);
+
+  {
+    const { relativePaths: tgzFilesRelative } = await getAllTgzPath();
+    console.log(`${TAG} Found ${tgzFilesRelative.length} .tgz files.`);
+    for (const filePath of tgzFilesRelative) {
+      await execCommand("attw", filePath);
+    }
+  }
+
+  function execCommand(binaryName: string, filePath: string): Promise<void> {
+    return new Promise((resolve) => {
+      const command = binaryName + " ./" + filePath;
+      exec(command, (error, stdout) => {
+        if (error) {
+          console.log(stdout);
+          attwFailedPackages.push(filePath);
+          resolve();
+        } else {
+          resolve();
+        }
+      });
+    });
+  }
+}
+
+if (esMain(import.meta)) {
+  await runAttwOnTgz();
+  if (attwFailedPackages.length > 0) {
```

Comment:
`attwFailedPackages` should be returned from the method itself and not updated by a function call as a side effect